File: DesignPatternDemo/Pattern/Command/Invoker.m
//
//  Invoker.m
//  DesignPatternDemo
//
//  Created by ruantong on 2019/5/11.
//  Copyright © 2019 wupeng. All rights reserved.
//

#import "Invoker.h"
#import "Command.h"

@interface Invoker ()
{
    Command * _command;
}
@end

@implementation Invoker

- (void)setCommand:(Command *)command{
    _command = command;
}

- (void)action{
    [_command execute];
}

@end

File: DesignPatternDemo/Pattern/Command/Invoker.swift
//
//  Invoker.swift
//  DesignPatternDemo
//
//  Created by wupeng on 2020/3/5.
//  Copyright © 2020 wupeng. All rights reserved.
//

import Foundation

class Invoker : NSObject {
    private var command:Command
    
    public init(command:Command) {
        self.command = command
    }
    
    func action() {
        self.command.execute()
    }
}
